packages/core/crates/core-strategy/src/promiscuous.rs

```diff
@@ -39,95 +48,161 @@ pub struct PromiscuousStrategy {
     /// Defaults to square-root of the sampled network size.
     pub max_channels: Option<usize>,
 
-    /// Determines if the strategy should automatically redeem tickets.
-    /// Defaults to false
-    pub auto_redeem_tickets: bool,
-
     /// If set, the strategy will aggressively close channels (even with peers above the `network_quality_threshold`)
     /// if the number of opened outgoing channels (regardless if opened by the strategy or manually) exceeds the
     /// `max_channels` limit.
     /// Defaults to true
     pub enforce_max_channels: bool,
-
-    sma: SimpleMovingAvg,
 }
 
-impl Default for PromiscuousStrategy {
+impl Default for PromiscuousStrategyConfig {
     fn default() -> Self {
-        PromiscuousStrategy {
+        PromiscuousStrategyConfig {
             network_quality_threshold: 0.5,
             new_channel_stake: Balance::from_str("100000000000000000", BalanceType::HOPR),
             minimum_channel_balance: Balance::from_str("10000000000000000", BalanceType::HOPR),
             minimum_node_balance: Balance::from_str("100000000000000000", BalanceType::HOPR),
             max_channels: None,
-            auto_redeem_tickets: true,
             enforce_max_channels: true,
-            sma: SimpleMovingAvg::new(),
         }
     }
 }
 
-impl ChannelStrategy for PromiscuousStrategy {
-    const NAME: &'static str = "promiscuous";
-
-    fn tick(
-        &mut self,
-        balance: Balance,
-        addresses: impl Iterator<Item = (Address, f64)>,
-        outgoing_channels: Vec<OutgoingChannelStatus>,
-    ) -> StrategyTickResult {
-        let mut to_open: Vec<OutgoingChannelStatus> = vec![];
-        let mut to_close: Vec<Address> = vec![];
+/// This strategy opens outgoing channels to peers, which have quality above a given threshold.
+/// At the same time, it closes outgoing channels opened to peers whose quality dropped below this threshold.
+pub struct PromiscuousStrategy<Db, Net>
+where
+    Db: HoprCoreEthereumDbActions,
+    Net: NetworkExternalActions,
+{
+    db: Arc<RwLock<Db>>,
+    network: Arc<RwLock<Network<Net>>>,
+    tx_sender: TransactionSender,
+    config: PromiscuousStrategyConfig,
+    sma: Arc<RwLock<SimpleMovingAvg>>,
```

Comment:
Does not need to be `Arc` unless you send it to a `spawn_local` or something. Just `RwLock` will be sufficient